src/parser/deserializer.ts

```diff
@@ -120,6 +123,47 @@ function deserializeObject(
   const promoteLongs = options['promoteLongs'] == null ? true : options['promoteLongs'];
   const promoteValues = options['promoteValues'] == null ? true : options['promoteValues'];
 
+  // Ensures default validation option if none given
+  const validation = options['validation'] == null ? { utf8: true } : options['validation'];
+
+  // Shows if global utf8 validation is enabled or disabled
+  let globalUTFValidation = true;
+  // Reflects utf8 validation boolean regardless of global or specific key validation
+  let uniformBool: boolean;
+  // Set of keys either to enable or disable validation on
+  const utf8KeysSet = new Set();
+
+  // Check for boolean uniformity and empty validation option
+  const keys = validation.utf8;
+  if (typeof keys !== 'boolean') {
+    globalUTFValidation = false;
+    const vals = Object.keys(keys).map(function (key) {
+      return keys[key];
+    });
+    if (vals.length !== 0) {
+      // Ensures boolean uniformity in utf-8 validation (all true or all false)
+      if (typeof vals[0] === 'boolean') {
+        uniformBool = vals[0];
+        if (!vals.every(item => item === uniformBool)) {
+          throw new BSONError(
+            'Invalid UTF-8 validation option - keys must be all true or all false'
+          );
+        }
+      } else {
+        throw new BSONError('Invalid UTF-8 validation option, must specify boolean values');
+      }
+    } else {
+      throw new BSONError('validation option is empty');
+    }
+  } else {
+    uniformBool = keys;
+  }
+
+  // Add keys to set that will either be validated or not based on uniformBool
+  if ((!uniformBool && !globalUTFValidation) || (uniformBool && !globalUTFValidation)) {
+    Object.keys(keys).forEach(key => utf8KeysSet.add(key));
```

Comment:
Super nit, in general we have preferred using js `for` loops as opposed to `.forEach`.